packages/plugins/slugManager/manager.ts

```diff
@@ -0,0 +1,158 @@
+import { IAPI, BaseEntity, Meta } from "./api.ts";
+interface ProfileUpdate {
+  meta: Meta;
+  isAlternateProfile: boolean;
+  [key: string]: string | boolean | Meta;
+}
+
+export interface InternalSlugManagerConfig {
+  slugFormat: string;
+  searchIds?: string[];
+  entityTypes?: string[];
+  slugField?: string;
+  slugFormatLocaleOverrides?: Record<string, string>;
+  api: Pick<
+    IAPI,
+    | "savedSearchesIncludeEntity"
+    | "updateField"
+    | "listLanguageProfiles"
+    | "listEntities"
+  >;
+}
+
+export function createManager(config: InternalSlugManagerConfig) {
+  const {
+    slugFormatLocaleOverrides = {},
+    searchIds = [],
```

Comment:
Will there be cases where people don't want to use a savedSearch but instead use the other scope types (entityType)?

Reply:
yes, the config also takes an `entityTypes` array: https://github.com/yext/pages/pull/210/files/d11d75dc9db5df837c0c8cd7377c4f6903034507#diff-9ebad0d98cc28c946d8f88e9fbc7b9b14aef410f3ea030940fa6e35e89fcb1cfR28

Reply:
I didn't format them as `scopes` because we aren't using streams here, it's using kapi which doesn't talk about filters the same way.